src/H5VLcallback.c

```diff
@@ -2060,21 +2110,44 @@ H5VL__dataset_read(void *obj, const H5VL_class_t *cls, hid_t mem_type_id, hid_t
  *-------------------------------------------------------------------------
  */
 herr_t
-H5VL_dataset_read(const H5VL_object_t *vol_obj, hid_t mem_type_id, hid_t mem_space_id, hid_t file_space_id,
-                  hid_t dxpl_id, void *buf, void **req)
+H5VL_dataset_read(size_t count, const H5VL_object_t *vol_obj[], hid_t mem_type_id[], hid_t mem_space_id[],
```

Comment:
It looks like there's only one place remaining in the feature branch where `H5VL_dataset_read` is used and it could probably be converted to the form that the new `H5VL_dataset_read_direct` is expected. It also looks like `H5VL_dataset_write` is no longer used, unless I missed a spot while searching. Based on that, does it seem worth it to just modify `H5VL_dataset_read` and `H5VL_dataset_write` rather than having two different internal H5VL I/O calls?

Reply:
I left those in for consistency with how other the other VOL callbacks work. We could maybe eliminate one of each function but I don't think the remaining functions should use the same naming convention as the other single underscore H5VL callback functions.

Reply:
Sure. I don't think this needs to hold up a merge but it's something to think about to lower some of the maintenance overhead.